### PR TITLE
linear_cross_entropy: test and document bias with probability targets

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.13.0) (oldest at bottom):
* __->__ #188433
* #188296

Probability (soft-label) targets combined with a linear bias already work on
the chunked path -- the loss adds the bias through the shared shifted_logits,
and the bias gradient is the batch-sum of the dense grad_logits the prob loop
already forms, correct for both the scalar and the reduction='none'
algorithms (which share that loop). Nothing rejected the combination, but it
had no test coverage and the docstring did not mention it. This adds the
coverage and the documentation; the operator is unchanged.

Correctness is checked against an fp64 reference (loss and input/weight/bias
gradients match to machine precision), guarded by the feps relative-error
bound. Per-element grad ULP caps are RNG-fragile drift trip-wires -- the
reason the sample set and its draw order are frozen -- and a bias makes them
heavier-tailed (the cancellation in softmax minus the target). Rather than
add new RNG-locked caps for the bias legs, the per-element grad ULP trip-wires
are skipped uniformly for all bias=True legs (index and probability alike),
which rely on the feps grad_error bound plus the output ULP instead; the
bias-free legs keep their calibrated caps. The one bias-specific tolerance is
fp32 prob+bias on hardware-matmul accelerators, whose input-grad relative
error rounds to ~3.8*eps (cpu's higher-precision matmul stays under 3*eps), so
it is given 5*eps.

The docstring now states the chunked path supports (C,)-shaped bias with
probability targets as well as class indices, and drops the reduction
enumeration for probability targets (mean/sum/none are all supported, so it
was no longer a restriction).

Stacked on #188296: the coverage spans reduction='none' alongside mean/sum.

Test Plan:

    CUDA_VISIBLE_DEVICES="" python test/test_nn.py -k linear_cross_entropy -v
    python test/test_nn.py -k linear_cross_entropy -v   # CUDA

Authored with the assistance of Claude (AI coding assistant).

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -15010,12 +15017,20 @@ def grad_max_ulp(grad, expected):
...
-        self.assertLessEqual(maximal_linear_bias_grad_max_ulp_diff, expected_linear_bias_grad_max_ulp_diff,
-                             msg=f"worst linear_bias-grad ULP {maximal_linear_bias_grad_max_ulp_diff} from kwargs={worst_linear_bias_grad_kwargs}")
+        # Per-element grad ULP caps are RNG-fragile drift trip-wires -- which
+        # is why the sample set and its draw order are frozen (a shift moves
+        # the calibrated caps). The bias=True legs (index and probability
+        # alike) instead rely on the feps grad_error bound for grad
+        # correctness plus the output ULP, and skip the per-element grad ULP
+        # trip-wires, so bias coverage adds no new RNG-locked caps. The
+        # bias-free legs keep their calibrated caps.
+        if not bias:
```
Comment (pearu) [inline test/test_nn.py:14172]: FYI, broader than the title: this skips the per-element grad ULP asserts for all `bias=True` legs, including the pre-existing index+bias ones -- not only the new probability legs. Rationale in the comment above + the commit message.

**Comment by slayton58** [comment]
@claude perform a deep review on this PR, concentrating on subtle errors that a human reviewer might miss. Only surface significant issues, ignore anything small. Reply in a collapsed comment.

**Comment by pearu** [comment]
Addressed in the latest pushes.

**Issue 1 (dead bias caps + retired bias-grad trip-wire):** restructured so `bias=True` is feps-only by construction. Removed all bias-specific per-element ULP cap literals (the `if bias:` accurate branches and the `N if bias else 0` compact/None lines), removed `expected_linear_bias_grad_max_ulp_diff` and its now-vacuous assert entirely, and gated the input/weight `maximal_*` per-element computation under `if not bias`. The three `grad_error` asserts run for every leg and are the bias correctness guard. This declines new caps for prob+bias and retires the pre-existing index+bias per-element caps, as an explicit documented decision; nothing changes in what is asserted for `bias=False`.

**Issue 2 (the prob+bias fp32 accelerator grad_error tolerance):** you were right that it was fragile -- CI on the CUDA-13 L4 shards proved it. The softmax-minus-target cancellation, amplified by the bias shift toward one-hot, inflates *all three* gradient relative errors there (input ~10.6*eps, weight ~7.9*eps; A100/CUDA12 was ~3.8*eps), while CPU stays under 3*eps and `bias=False` stays at feps on the same hardware. It's legitimate IEEE-fp32 (TF32 is off for this test; ~1e-6 magnitude, not TF32's ~1e-3) -- a cuBLAS K-accumulation-order difference across versions, not a bug. So the input/weight/bias `grad_error` bounds now share one 16*eps tolerance for these legs (covering the observed values with margin); everything else stays at feps.

_Drafted by Claude Code (an AI agent) and reviewed & approved by pearu._

**Comment by slayton58** [comment]
@claude perform a deep review in a collapsed comment, concentrating on making sure there are no unintended consequences of the changes. Only surface significant issues, ignore anything small.

**Comment by pearu** [comment]
Both points from the review are addressed in the latest push:

- Issue 2 (scope): weight/bias `grad_error` are back at feps for every leg; only the prob+bias+fp32+accelerator legs take the 16*eps widening (via a separate `wb_grad_err_tol`), so no `bias=False` leg's tolerance changes.
- Issue 1 (coverage): the acc_dtype prob+bias sample now forces >=2 chunks, so the mixed-precision bias-grad chunk-boundary commit is exercised on the probability path too.

Test + doc only, operator unchanged. The CUDA prob+bias shard that previously failed is green; the only red is an unrelated ROCm-mi350 distributed timeout (`test_grads_same_across_ranks_with_no_sync`, exit 124 -- no LCE test runs in that shard), being re-run.

---
_Drafted by Claude Code (an AI agent) and reviewed & approved by pearu._

**Review by slayton58 — APPROVED** [review]
